oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -274,16 +284,35 @@ private static boolean hasAnyHiddenNodes(NodeBuilder builder){
         return false;
     }
 
+    private void sparseLoggger(String primaryType, String name) {
+        synchronized (countLock) {
+            if ((cyclicExecutionCount >= indexJcrTypeInvalidLogLimiter)) {
```

Comment:
I liked the previous "executionCount % LIMIT == 0" better, because it's simpler.

Reply:
well yes, it is complicated than executionCount % LIMIT but I think it is not that complicated. We are just resetting the count after LIMIT.

Reply:
having executionCount not going back to 0 would give us some advantage I think... we could log the value, and therefore see how often the method was called... but it's not directly needed.